__tests__/app.test.js

```diff
@@ -63,10 +63,80 @@ describe("/api/articles", () => {
         .expect(200)
         .then(({ body }) => {
           const articleArr = body.articles;
-
           expect(articleArr).toBeSortedBy("created_at", { descending: true });
         });
     });
+
+    test("200: Should respond with filtered results by topic when passed a topic query", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200)
+        .then(({ body }) => {
+          const articleArr = body.articles;
+
+          expect(Array.isArray(articleArr)).toBe(true);
+          articleArr.forEach((topic) => {
+            expect(topic).toHaveProperty("topic", "mitch");
+          });
+        });
+    });
+
+    test("200: Should respond with sorted by any valid category", () => {
+      return request(app)
+        .get("/api/articles?sort_by=votes")
+        .expect(200)
+        .then(({ body }) => {
+          const articleArr = body.articles;
+
+          expect(Array.isArray(articleArr)).toBe(true);
+          expect(articleArr).toBeSortedBy("votes", { descending: true });
+        });
+    });
+
+    test("200: Should respond with sorted by ascending or descending by query", () => {
+      return request(app)
+        .get("/api/articles?order=ASC")
+        .expect(200)
+        .then(({ body }) => {
+          const articleArr = body.articles;
+
+          expect(Array.isArray(articleArr)).toBe(true);
+          expect(articleArr).toBeSortedBy("created_at", { ascending: true });
+        });
+    });
+
+    test("200: Should be able to handle multiple queries", () => {
+      return request(app)
+        .get("/api/articles?order=ASC&sort_by=votes&topic=mitch")
+        .expect(200)
+        .then(({ body }) => {
+          const articleArr = body.articles;
+
+          expect(Array.isArray(articleArr)).toBe(true);
+          expect(articleArr).toBeSortedBy("votes", { ascending: true });
+          articleArr.forEach((topic) => {
+            expect(topic).toHaveProperty("topic", "mitch");
+          });
+        });
+    });
+
+    test("400: Should recieve bad request if query key is invalid", () => {
+      return request(app)
+        .get("/api/articles?oder=ASC")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Bad Request");
+        });
+    });
+
+    test("400: Should recieve bad request if query value is invalid", () => {
+      return request(app)
+        .get("/api/articles?order=AC")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Bad Request");
+        });
+    });
```

Comment:
some brilliant work here Cameron! There are a couple of tests that just need adding. 

400 - invalid sort_by query - ?sort_by=bananas 
404 - valid but none existent topic